### PR TITLE
mgr/rook: Support other system namespaces

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes http://tracker.ceph.com/issues/38799

Refactor is required for https://github.com/rook/rook/pull/2820

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/rook/module.py [hunk 1]
```diff
@@ -134,6 +134,20 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class RookEnv(object):
+    def __init__(self):
+        pod_namespace = os.environ.get('POD_NAMESPACE', 'rook-ceph')
+        self.cluster_name = os.environ.get('ROOK_CLUSTER_NAME', pod_namespace)
+        self.cluster_ns = os.environ.get('ROOK_CLUSTER_NS', self.cluster_name)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/rook/module.py]: todo: remove ROOK_CLUSTER_NS

**Comment by yuriw** [comment]
@theanalyst @smithfarm FYI requested `nautilus` backport for this

File: src/pybind/mgr/rook/module.py [hunk 2]
```diff
@@ -134,6 +134,20 @@ def wrapper(*args, **kwargs):
...
+
+        self.system_ns = os.environ.get('ROOK_SYSTEM_NS', "rook-ceph-system")
```
Comment (BlaineEXE) [inline src/pybind/mgr/rook/module.py]: Can we change this to `ROOK_OPERATOR_NAMESPACE`

File: src/pybind/mgr/rook/module.py [hunk 3]
```diff
@@ -134,6 +134,20 @@ def wrapper(*args, **kwargs):
...
+        self.api_version = os.environ.get('ROOK_API_VERSION', 'v1')
```
Comment (BlaineEXE) [inline src/pybind/mgr/rook/module.py]: Can we change this to `ROOK_CEPH_CRD_VERSION`

File: src/pybind/mgr/rook/module.py [hunk 4]
```diff
@@ -134,6 +134,20 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class RookEnv(object):
+    def __init__(self):
+        pod_namespace = os.environ.get('POD_NAMESPACE', 'rook-ceph')
+        self.cluster_name = os.environ.get('ROOK_CLUSTER_NAME', pod_namespace)
```
Comment (rohan47) [inline src/pybind/mgr/rook/module.py]: This has been renamed to `ROOK_CEPH_CLUSTER_CRD_NAME` in https://github.com/rook/rook/pull/2903

File: src/pybind/mgr/rook/module.py (hunk 3 above)
Comment (rohan47) [reply src/pybind/mgr/rook/module.py]: This has been renamed to `ROOK_CEPH_CLUSTER_CRD_VERSION` in https://github.com/rook/rook/pull/2903

Comment (sebastian-philipp) [reply src/pybind/mgr/rook/module.py]: Nah, I'm going to wait for another day or two, till I rename them again.

File: src/pybind/mgr/rook/module.py (hunk 4 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/rook/module.py]: Nah, I'm going to wait for another day or two, till I rename them again.

Comment (rohan47) [reply src/pybind/mgr/rook/module.py]: Yup, that sounds good, Please wait till it gets merged.

**Comment by travisn** [comment]
https://github.com/rook/rook/pull/2903 has been merged so we can assume those are the final env var names and go ahead with this.

File: src/pybind/mgr/rook/module.py (hunk 1 above)
Comment (travisn) [reply src/pybind/mgr/rook/module.py]: I believe `self.cluster_ns` should be set to `pod_namespace`. The namespace could be different from the cluster CR name and `cluster_ns` represents the namespace where the mgr is running, right?

Comment (sebastian-philipp) [reply src/pybind/mgr/rook/module.py]: fixed

Comment (sebastian-philipp) [reply src/pybind/mgr/rook/module.py]: fixed

**Comment by sebastian-philipp** [comment]
jenkins retest this please

**Comment by sebastian-philipp** [comment]
jenkins retest this please

**Pull request merged**